tests/unit/test_mesmer_x_distrib_cov.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import pytest
+
+from mesmer.mesmer_x import Expression, distrib_cov
+
+
+def test_distrib_cov_init_all_default():
+    rng = np.random.default_rng(0)
+    n = 250
+    pred = np.linspace(0, 1, n)
+    targ = rng.normal(loc=2 * pred, scale=0.1, size=n)
+
+    expression = Expression("norm(loc=c1 * __tas__, scale=c2)", expr_name="exp1")
+
+    dist = distrib_cov(targ, {"tas": pred}, expression)
+
+    np.testing.assert_equal(dist.data_targ, targ)
+    np.testing.assert_equal(dist.data_pred, {"tas": pred})
+    np.testing.assert_equal(dist.weights_driver, np.ones(n) / n)
+    assert dist.n_sample == n
+    assert dist.expr_fit == expression
+    assert not dist.add_test
+    assert dist.data_targ_addtest is None
+    assert dist.data_preds_addtest is None
+    assert dist.threshold_min_proba == 1e-09
+    assert dist.boundaries_params == expression.boundaries_parameters
+    assert dist.boundaries_coeffs == {}
+    assert dist.first_guess is None
+    assert dist.func_first_guess is None
+    assert dist.n_coeffs == 2
+    assert dist.scores_fit == ["func_optim", "NLL", "BIC"]
+    assert dist.xtol_req == 1e-06
+    assert dist.ftol_req == 1e-06
+    assert dist.maxiter == 1000 * dist.n_coeffs * np.log(dist.n_coeffs)
+    assert dist.maxfev == 1000 * dist.n_coeffs * np.log(dist.n_coeffs)
+    assert dist.method_fit == "Powell"
+    assert dist.name_ftol == "ftol"
+    assert dist.name_xtol == "xtol"
+    assert not dist.error_failedfit
+    assert not dist.fg_with_global_opti
+    assert not dist.weighted_NLL
+    assert dist.type_fun_optim == "NLL"
+    assert dist.threshold_stopping_rule is None
+    assert dist.exclude_trigger is None
+    assert dist.ind_year_thres is None
+
+
+def test_distrib_cov_init():
+    rng = np.random.default_rng(0)
+    n = 250
+    pred = np.linspace(0, 1, n)
+    targ = rng.normal(loc=2 * pred, scale=0.1, size=n)
+
+    expression = Expression("norm(loc=c1 * __tas__, scale=c2)", expr_name="exp1")
+
+    data_targ_addtest = rng.normal(loc=2 * pred, scale=0.1, size=n)
+    data_preds_addtest = {"tas": np.linspace(0, 0.9, n)}
+    threshold_min_proba = 0.1
+    boundaries_params = {"loc": [-10, 10], "scale": [0, 1]}
```

Comment:
```suggestion
    boundaries_params = {"loc": [-10, 10], "scale": [-1, 1]}
```

Could try -1 here to check we end up with 0 again